tests/app/questionnaire/test_list_action.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+from mock import Mock, MagicMock, patch
+
+from app.views.handlers.list_action import ListAction
+from app.questionnaire.location import InvalidLocationException
+from app.questionnaire.routing_path import RoutingPath
+from app.questionnaire.location import Location
+from app.questionnaire.router import Router
+from app.data_model.answer_store import AnswerStore
+from app.data_model.list_store import ListStore
+from app.data_model.progress_store import ProgressStore, CompletionStatus
+from tests.app.app_context_test_case import AppContextTestCase
+
+
+class TestListAction(AppContextTestCase):
+    def setUp(self):
```

Comment:
I think this setUp method needs to be tidied, some of the bits in the set up seem like they are only relevant to the test below

Reply:
I had to keep some of them where they are because of linting, but unnecessary attributes were removed.👍